api/src/main/java/org/xwiki/apps/googleapps/internal/GoogleAppsManagerImpl.java

```diff
@@ -0,0 +1,1372 @@
+ * @version $Id$
+ * @since 3.0
+ */
+@Component
+@Singleton
+public class GoogleAppsManagerImpl
+        implements GoogleAppsManager, EventListener, Initializable, Disposable
+{
+
+    //  -----------------------------  Lifecycle ---------------------------
+
+    @Inject
+    private Provider<XWikiContext> xwikiContextProvider;
+
+    @Inject
+    private QueryManager queryManager;
+
+    @Inject
+    private Environment environment;
+
+    @Inject
+    @Named("current")
+    private DocumentReferenceResolver<String> documentResolver;
+
+    @Inject
+    private Logger log;
+
+    @Inject
+    private ComponentManager componentManager;
+
+    @Override
+    public String getName()
+    {
+        return "googleapps.scriptservice";
+    }
+
+
+    @Override
+    public void initialize() throws InitializationException
+    {
+        log.info("GoogleAppsScriptService initting.");
+        XWiki xwiki = getXWiki();
+        XWikiContext context = xwikiContextProvider.get();
+
+        if (context != null) {
+            readConfigDoc(context);
+        }
+
+        if (xwiki != null && context != null) {
+            log.info("Initting authService.");
+            // We do not verify with the context if the plugin is active and if the license is active
+            // this will be done by the GoogleAppsAuthService and UI pages later on, when it is called within a request
+            try {
+                authService = componentManager.getInstance(GoogleAppsAuthService.class);
+                authService.setGoogleAppsManager(this);
+                xwiki.setAuthService(authService);
+                log.info("Succeeded initting authService,");
+            } catch (ComponentLookupException e) {
+                log.info("Failed initting authService", e);
+            }
+        }
+        if (authService == null) {
+            log.info("Not yet initting authService.");
+        }
+
+        try {
+            jacksonFactory = JacksonFactory.getDefaultInstance();
+            httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+        } catch (Exception  e) {
+            e.printStackTrace();
+            throw new InitializationException("Trouble at initializing", e);
+        }
+    }
+
+    @Override
+    public List<Event> getEvents()
+    {
+        return Arrays.asList(new ApplicationReadyEvent(), new DocumentUpdatedEvent());
+    }
+
+    @Override
+    public void onEvent(Event event, Object source, Object data)
+    {
+        log.info("Event triggered: " + event + " with source " + source);
+        boolean applicationStarted = false, configChanged = false;
+        if (event instanceof ApplicationReadyEvent) {
+            applicationStarted = true;
+        }
+        if (event instanceof DocumentUpdatedEvent) {
+            XWikiDocument document = (XWikiDocument) source;
+            configDocRef = getConfigDocRef();
+            if (document != null && document.getDocumentReference().compareTo(configDocRef) == 0) {
```

Comment:
Same here: hard to imagine when the document could be null if the event is triggered in case of document update.

Reply:
I agree. Can we have a safety here? Is there an assertion?